src/OpenTelemetry.Instrumentation.SqlClient/SqlClientInstrumentationOptions.cs

```diff
@@ -302,23 +309,45 @@ internal void AddConnectionLevelDetailsToActivity(string dataSource, Activity sq
                     ConnectionDetailCache.TryAdd(dataSource, connectionDetails);
                 }
 
-                if (!string.IsNullOrEmpty(connectionDetails.ServerHostName))
-                {
-                    sqlActivity.SetTag(SemanticConventions.AttributeNetPeerName, connectionDetails.ServerHostName);
-                }
-                else
+                if (!string.IsNullOrEmpty(connectionDetails.InstanceName))
```

Comment:
Not related to this PR, but why is this method in the options class instead of the listener?

We should probably move this code to the listener. That would also make us consistent with other instrumentation libraries for how we read the environment variable.